test/Microsoft.DotNet.ShellShim.Tests/ShellShimManagerTests.cs

```diff
@@ -0,0 +1,436 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Transactions;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.DotNet.Tools;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.DotNet.Tools.Tests.ComponentMocks;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.ShellShim.Tests
+{
+    public class ShellShimManagerTests : TestBase
+    {
+        private readonly ITestOutputHelper _output;
```

Comment:
Could you mark all the new tests?

Reply:
Tests added:

* `GivenANonexistentShimRemoveDoesNotThrow`
* `GivenAnInstalledShimRemoveDeletesTheShimFiles`
* `GivenAnInstalledShimRemoveRollsbackIfTransactionIsAborted`
* `GivenAnInstalledShimRemoveCommitsIfTransactionIsCompleted`